manager/src/main/java/com/redhat/service/smartevents/manager/api/user/SchemaAPI.java

```diff
@@ -0,0 +1,82 @@
+import org.eclipse.microprofile.openapi.annotations.security.SecurityRequirement;
+import org.eclipse.microprofile.openapi.annotations.security.SecurityScheme;
+import org.eclipse.microprofile.openapi.annotations.security.SecuritySchemes;
+import org.eclipse.microprofile.openapi.annotations.tags.Tag;
+
+import com.redhat.service.smartevents.infra.api.APIConstants;
+import com.redhat.service.smartevents.manager.api.models.responses.ProcessorCatalogResponse;
+import com.redhat.service.smartevents.manager.api.models.responses.ProcessorSchemaEntryResponse;
+
+import io.quarkus.security.Authenticated;
+
+@Tag(name = "Schema Catalog", description = "The API that provide the JSON schemas for the actions/sources processors")
+@SecuritySchemes(value = {
+        @SecurityScheme(securitySchemeName = "bearer",
+                type = SecuritySchemeType.HTTP,
+                scheme = "Bearer")
+})
+@SecurityRequirement(name = "bearer")
+@Path(APIConstants.SCHEMA_API_BASE_PATH)
+@Consumes(MediaType.APPLICATION_JSON)
+@Produces(MediaType.APPLICATION_JSON)
+@Authenticated
+public class SchemaAPI {
+
+    private static final String ACTIONS_DIR_PATH = "/schemas/actions/";
+    private static final String SOURCES_DIR_PATH = "/schemas/sources/";
+    private static final String JSON_FILE_EXTENSION = ".json";
+    private static final String ACTION_TYPE = "action";
+    private static final String SOURCE_TYPE = "source";
+
+    private List<String> sinks;
+    private List<String> sources;
+
+    @PostConstruct
+    void init() throws IOException {
+        sinks = IOUtils.readLines(getClass().getResourceAsStream(ACTIONS_DIR_PATH), StandardCharsets.UTF_8);
+        sources = IOUtils.readLines(getClass().getResourceAsStream(SOURCES_DIR_PATH), StandardCharsets.UTF_8);
+    }
+
+    @APIResponses(value = {
+            @APIResponse(description = "Success.", responseCode = "200",
+                    content = @Content(mediaType = MediaType.APPLICATION_JSON, schema = @Schema(implementation = ProcessorCatalogResponse.class))),
+            @APIResponse(description = "Bad request.", responseCode = "400", content = @Content(mediaType = MediaType.APPLICATION_JSON)),
+            @APIResponse(description = "Unauthorized.", responseCode = "401"),
+            @APIResponse(description = "Forbidden.", responseCode = "403"),
+            @APIResponse(description = "Internal error.", responseCode = "500", content = @Content(mediaType = MediaType.APPLICATION_JSON))
+    })
+    @Operation(summary = "Get a Processor of a Bridge instance", description = "Get a Processor of a Bridge instance for the authenticated user.")
+    @GET
+    public Response getCatalog() {
+        List<ProcessorSchemaEntryResponse> entries = new ArrayList<>();
+        entries.addAll(
+                sinks.stream().map(x -> new ProcessorSchemaEntryResponse(x.replace(JSON_FILE_EXTENSION, ""), ACTION_TYPE, APIConstants.SOURCES_SCHEMA_API_BASE_PATH + x)).collect(Collectors.toList()));
```

Comment:
The files provided by MC aren't valid json schema. We should parse only the JsonSchema part and hopefully validate them see 

https://github.com/5733d9e2be6485d52ffa08870cabdee0/sandbox/pull/771/files#diff-72d51b916df916a284e344a4fc442fe944c5e45a809e17ababd7da7407283865R20

Reply:
MGDOBR-765 is only for the endpoint of the available actions/sources. I'm not using the content of such files at all: they will be populated in 766 and an endpoint to retrieve their content will be implemented